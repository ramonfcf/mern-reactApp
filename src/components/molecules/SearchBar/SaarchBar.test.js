import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

import SearchBar from "./SearchBar";

describe("SearchBar component", () => {
  const onChange = jest.fn();
  const placeholder = "Test placeholder";

  it("matches snapshot", () => {
    const { container } = render(
      <SearchBar onChange={onChange} placeholder={placeholder} value="" />
    );
    expect(container).toMatchSnapshot();
  });

  it("calls onChange prop when input value changes", () => {
    render(
      <SearchBar onChange={onChange} placeholder={placeholder} value="" />
    );
    const inputElement = screen.getByRole("textbox");

    fireEvent.change(inputElement, { target: { value: "New value" } });
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it("renders with correct placeholder", () => {
    render(
      <SearchBar onChange={onChange} placeholder={placeholder} value="" />
    );
    const inputElement = screen.getByPlaceholderText(placeholder);

    expect(inputElement).toHaveAttribute("placeholder", placeholder);
  });
});
