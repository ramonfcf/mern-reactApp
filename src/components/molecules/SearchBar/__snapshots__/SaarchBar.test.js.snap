// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component matches snapshot 1`] = `
<div>
  <div
    class="input-group my-3"
  >
    <span
      class="input-group-text"
    >
      Search
    </span>
    <input
      class="form-control form-control-sm"
      placeholder="Test placeholder"
      type="text"
      value=""
    />
  </div>
</div>
`;
